py-polars/polars/io/spreadsheet/functions.py

```diff
@@ -421,27 +421,45 @@ def _read_spreadsheet(
     reader_fn, parser, worksheets = _initialise_spreadsheet_parser(
         engine, source, engine_options or {}
     )
-
-    # determine which named worksheets to read
-    if sheet_id is None and sheet_name is None:
-        sheet_names = [worksheets[0]["name"]]
-        return_multi = False
-    else:
-        return_multi = (
-            sheet_id == 0
-            or isinstance(sheet_id, Sequence)
-            or (isinstance(sheet_name, Sequence) and not isinstance(sheet_name, str))
-        )
-        ids = (sheet_id,) if isinstance(sheet_id, int) else sheet_id or ()
-        names = (sheet_name,) if isinstance(sheet_name, str) else sheet_name or ()
-        sheet_names = [
-            ws["name"]
-            for ws in worksheets
-            if (sheet_id == 0 or ws["index"] in ids or ws["name"] in names)
-        ]
-
-    # read data from the indicated sheet(s)
     try:
+        # determine which named worksheets to read
```

Comment:
Moving so much code into a single `try` block seems problematic 🤔 maybe extract into a function?

Reply:
Good plan; that block has just doubled in size... will factor out ✌️